packages/core/admin/admin/src/content-manager/pages/EditSettingsView/components/ModalForm.tsx

```diff
@@ -116,7 +126,7 @@ const ModalForm = ({ onMetaChange, onSizeChange }) => {
             value={fieldForm?.size}
             name="size"
             onChange={(value) => {
-              onSizeChange({ name: selectedField, value });
+              onSizeChange({ name: selectedField, value: value as number });
```

Comment:
```suggestion
/**
 * TODO: refactor this to be actually typesafe instead of casting.
 */
              onSizeChange({ name: selectedField, value: value as number });
```